### PR TITLE
KeyVaultSecretsRepository will now page through all secrets in Key Vault

### LINKED ISSUE #6031: All Function Keys Regenerate Constantly
Investigative information
Please provide the following:

**Timestamp**:
**Function App version**: 2.0.13351.0
**Function App name**: automation-lnpojzrfgkuqy
**Function name(s) (as appropriate)**: function invocation not required
**Invocation ID**: function invocation not required
**Region**: East US

## Repro steps
I have a function app, which any time any of the function keys pages are loaded or key is used by performing an API call, such as to an HTTP trigger, the keys are immediately regenerated, which is preventing the calls from being successful. 

For instance, if I go to this website https://portal.azure.com/#{DOMAIN}/resource/subscriptions/{SUBSCRIPTION}/resourceGroups/{RESOURCE_GROUP}/providers/Microsoft.Web/sites/{SITENAME}/functionsAppKeys and view the function keys, I'll see one version of the key. I can then reload the page and see another value of the function of the key right away. I can also view the keys refreshing if I just hit the the refresh button.

## Expected behavior
When refreshing the page, I expect that the host function/master keys should not change w/o explicitly renewing them

## Actual behavior
The host function/master key or the function key of any implemented function renews anytime they are accessed (viewed or a function is called which attempts to authenticate with them)

## Known workarounds
none, I can't get the function keys to stop regenerating

## Related information
Here are some possibly relevant facts to my setup with the function key management

- AzureWebJobsSecretStorageType = keyvault
  - The access to the kevault is working, when I view the keys, can I go look at the head version of the secret in the keyvault and see the same value. Then; when reloading the page, the backend keyvault setting is also being update.
- Function App is an I1 SKu in an ILB App Service Environment, DNS & SSL certs are setup so the function host is available from another machine on a peered vnet. E.g. the kudu console is available
  - There are 2 instances of the azure function apps running. But they are in different app service plans under different App Service Environments, within the same Azure Region.  They are running the exact same version of code and share the same configuration (think DR instances as a warm spare that is ready to be switch in if the primary app service environment failed)
- FUNCTIONS_WORKER_RUNTIME = dotnet
- FUNCTIONS_EXTENSION_VERSION = ~2

### PR DESCRIPTION
**Pull request by gzuber**
### Issue describing the changes in this PR

resolves #6031

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [x] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Security/KeyManagement/KeyVaultSecretsRepository.cs [hunk 1]
```diff
@@ -195,6 +203,40 @@ private async Task<ScriptSecrets> ReadFunctionSecrets(string functionName)
             return functionSecrets;
         }
 
+        private async Task<List<IPage<SecretItem>>> GetSecretsPagesAsync()
```
Comment (alrod) [inline src/WebJobs.Script.WebHost/Security/KeyManagement/KeyVaultSecretsRepository.cs]: Can we add some tests for the pagination?

Comment (alrod) [reply src/WebJobs.Script.WebHost/Security/KeyManagement/KeyVaultSecretsRepository.cs]: E2E tests:
https://github.com/Azure/azure-functions-host/blob/dev/test/WebJobs.Script.Tests.Integration/Host/SecretsRepositoryTests.cs#L50
https://github.com/Azure/azure-functions-host/blob/dev/test/WebJobs.Script.Tests.Integration/Host/SecretsRepositoryTests.cs#L87

Unit tests: (we can add one for FindSecrets)
https://github.com/Azure/azure-functions-host/blob/dev/test/WebJobs.Script.Tests/Security/KeyVaultSecretsRepositoryTests.cs

**Review by alrod — APPROVED** [review]

File: test/WebJobs.Script.Tests.Integration/Host/SecretsRepositoryTests.cs [hunk 2]
```diff
@@ -134,6 +135,78 @@ public async Task ReadAsync_ReadsExpectedFile(SecretsRepositoryType repositoryTy
             }
         }
 
+        [Theory] // Only for Key Vault to test paging over large number of secrets
+        [InlineData(SecretsRepositoryType.KeyVault, ScriptSecretsType.Host)]
+        [InlineData(SecretsRepositoryType.KeyVault, ScriptSecretsType.Function)]
+        public async Task ReadAsync_ReadsExpectedLargeFile(SecretsRepositoryType repositoryType, ScriptSecretsType secretsType)
```
Comment (alrod) [inline test/WebJobs.Script.Tests.Integration/Host/SecretsRepositoryTests.cs]: `ReadAsync_ReadsExpectedLargeFile `? Maybe `ReadAsync_ReadsExpectedKeyVaultPages`?

**Review by alrod — DISMISSED** [review]
Approved with a comment.

**Review by alrod — DISMISSED** [review]

**Review by yojagad — APPROVED** [review]

**Pull request merged**